app/services/analysis/gwas/genotype_vcf_to_csv_converter.rb

```diff
@@ -52,6 +35,79 @@ def call(path)
 
       private
 
+      # Extract mutation and samples data from a single VCF record. If VCF record
+      # specifies more than one alternative base it needs to be mapped to more
+      # than one CSV column.
+      #
+      # If there is no variation (i.e. there is only one distinct value apart
+      # from NA for each sample) for given mutation then it is silently skipped.
+      #
+      # Special characters apart from underscore are stripped from sample and
+      # mutation identifiers.
+      #
+      # record - VCF record
+      # sample_ids - original VCF sample identifiers
+      #
+      # Returns a hash containing mutation names, positions and
+      #   values for each sample.
+      def process_record(record, sample_ids)
+        mutation_names = []
+        mutation_data = []
+        sample_data = Hash.new { |data, sample_name| data[sample_name] = [] }
+
+        record.alt.each.with_index do |alternative, idx|
+          alternative_no = idx + 1
+          sample_values = []
+
+          sample_ids.each do |sample_name|
+            sample = record.sample_by_name(sample_name)
+            value = sample_value(sample, alternative_no)
+
+            sample_values << [sample_name, value]
+          end
+
+          unique_values = sample_values.map { |_, val| val }.uniq
+
+          if (unique_values - ["NA"]).size > 1
+            mutation_names << "#{record.id}_#{record.ref}_#{alternative}".strip.gsub(/\W/, '_')
+            mutation_data << [record.chrom.to_s.strip, record.pos.to_s.strip]
+
+            sample_values.each { |sample_name, val| sample_data[sample_name] << val }
+          end
+        end
+
+        {
+          mutation_names: mutation_names,
+          mutation_data: mutation_data,
+          sample_data: sample_data
+        }
+      end
+
+      # Find number of alternative alleles for given sample.
+      #
+      # sample - sample data (including position in genotype)
+      # alternative_no - ordinal number of alternative base (1-based,
+      #
+      # Returns:
+      #   2 for alternative allele in each chromosome
+      #   1 for alternative allele in any chromosome
+      #   0 for no mutation
+      #   NA if call could not be made
+      def sample_value(sample, alternative_no)
+        if sample.empty?
```

Comment:
Maybe `return "NA" if sample.empty?` (also remove the comment, since it is already explained above the signature).